src/classes/LVL_Level_TDTM.cls

```diff
@@ -0,0 +1,166 @@
+                    
+                    // go thru our new/updated levels
+                    for (integer i = 0; i < newlist.size(); i++) {
+                        Level__c lvl = (Level__c)newlist[i];
+                        
+                        // add new levels to their ladder
+                        if (lvl.Id == null) {
+                            if (lvl.Active__c) {
+                                list<Level__c> listLvl = getListLevel(mapLadderToLevels, lvl.Level_Field__c);
+                                insertLevel(lvl, listLvl);
+                            }
+                        }
+                        
+                        // deal with updates, by removing from old ladder and adding to new ladder
+                        if (lvl.Id != null) {
+                            
+                            // remove from old ladder
+                            Level__c lvlOld = (Level__c)oldlist[i];
+                            list<Level__c> listLvlOld = getListLevel(mapLadderToLevels, lvlOld.Level_Field__c);
+                            removeLevel(lvlOld, listLvlOld);
+                            
+                            // add to new ladder
+                            if (lvl.Active__c) {
+                                list<Level__c> listLvl = getListLevel(mapLadderToLevels, lvl.Level_Field__c);
+                                insertLevel(lvl, listLvl);
+                            }
+                        }
+                    }
+                    
+                    // now validate that each ladder has no overlaps
+                    for (list<Level__c> listLvl : mapLadderToLevels.values()) {
+                        verifyNoOverlapListLvl(listLvl);
+                    }
+                }
+            }
+        return null;
+    }
+    
+    /*******************************************************************************************************
+    * @description get the list of Levels for the given ladder, creating the list if empty.
+    * @param mapLadderToLevels The map to get the list from
+    * @param strLadder The key into the map
+    * @return list<Level__c> the List
+    */
+    private static list<Level__c> getListLevel(map<String, list<Level__c>> mapLadderToLevels, string strLadder) {
+        list<Level__c> listLvl = mapLadderToLevels.get(strLadder);
+        if (listLvl == null) {
+            listLvl = new list<Level__c>();
+            mapLadderToLevels.put(strLadder, listLvl);
+        }
+        return listLvl;
+    }
+    
+    /*******************************************************************************************************
+    * @description insert the Level in the correct order for the list of Levels
+    * @param lvl The Level to insert
+    * @param listLvl The list to insert into
+    */
+    private void insertLevel(Level__c lvl, list<Level__c> listLvl) {
```

Comment:
I think we should call this "addLevel" to correspond with the list methods, insert makes it feel like it's performing DML.